### ISSUE TITLE
Tabset Feature Improvement: Allow the select command to use object, for IDs

### ISSUE THREAD

**Issue by Adondriel** [ng-bootstrap/ng-bootstrap#1682]
Feature: The ID already supports setting it to an Object, so I'de like to be able to select the tab, via an Object, rather than just a string.

I will be trying to solve this in my fork, I just wanted to create ticket here, so that I can PR it with some kind of ticket to reference, once I'm done.
### Version of Angular, ng-bootstrap, and Bootstrap:

Angular: 4.2.3

ng-bootstrap: 1.0.0-alpha.26

Bootstrap: 4.0.0-alpha6

**Comment by pkozlowski-opensource** [comment]
The `id` property of a tab is documented as `string` here: https://ng-bootstrap.github.io/#/components/tabs/api#NgbTab and if we don't stringify I would say that this is a bug rather than intentional behaviour... 

Why would you like to use sth else than string? Remember that those IDs are used as an `id` property / attribute in the DOM and those are essential for proper accessibility support.

tl;dr; is that I'm not keen on allowing anything else than strings unless there are _very_ good reasons for it _and_ accessibility is addressed properly.

**Comment by Adondriel** [comment]
Technically this is, a bug, but the reason I'm doing this is because I do this: 

```
    selectCostbook(event: any) {
        // This is because we are able to store the object, in the [id] field of the tab, so that we don't have to re-fetch the costbook, for each tab change.
        this.currentCostbook = event.nextId;
        this.isEditing = false;
    }
```
selectCostbook() being the event that changing tabs sends, and my tabset looks like: 
```
    <ngb-tabset #t="ngbTabset" (tabChange)="selectCostbook($event)">
      <ngb-tab *ngFor="let c of costbooks" [id]="c">
        <ng-template ngbTabTitle>{{c?.costbookName}}</ng-template>
        <ng-template ngbTabContent>
          <app-costbookconfig [currentCostbook]="currentCostbook" [isEditing]="isEditing" [projectId]="projectId" [isOnProjectPage]="true" (newCostbook)="createNewCostbook()"></app-costbookconfig>
        </ng-template>
      </ngb-tab>
    </ngb-tabset>
```

**Comment by Adondriel** [comment]
I've updated my code, to fetch via ID, everything is working, except for: 
```
    createNewCostbook() {
        this.currentCostbook = new Costbook();
        this.currentCostbook.isNew = true;
        this.isEditing = true;
        this.currentCostbook.costbookId = -1;
        this.currentCostbook.costbookName = 'New Costbook';
        this.costbooks.push(this.currentCostbook);
        this.t.select(this.currentCostbook.costbookId);
    }
```
Could this be due to the tab not existing yet? Like, is it too soon after I'm modifying the array of tabs, and maybe it hasn't been created yet?

**Comment by Adondriel** [comment]
Since this isn't really needed anymore, I'll close this issue... unless you want me to keep it here, so that we can fix the thing with Id's being objects.
